### ISSUE TITLE
zsh: segmentation fault

### ISSUE THREAD

**Issue by giovannividetta** [MIC-DKFZ/TractSeg#269]
Hello, everyone!

I am trying to apply TractSeg on my diffusion MRI data. In order to understand the algorithm, I am running TractSeg on the example data provided by @wasserth in his GitHub repository. Launching on terminal **TractSeg -i Diffusion.nii.gz --raw_diffusion_input**, a segfault error appears as following:

_Creating brain mask...
Warning: An input intended to be a single 3D volume has multiple timepoints. Input will be truncated to first volume, but this functionality is deprecated and will be removed in a future release.
Creating peaks (1 of 3)...
Creating peaks (2 of 3)...
Creating peaks (3 of 3)...
Loading weights from: /Users/brainmilab/.tractseg/pretrained_weights_tract_segmentation_v3.npz
Processing direction (1 of 3)
zsh: segmentation fault  TractSeg -i Diffusion.nii.gz --raw_diffusion_input_

I looked for a solution on websites and forums, but I did not find any of relevant. I downloaded pretrained_weights_tract_segmentation_v3.npz manually from zenodo (https://zenodo.org/records/5052821) because I thought that file was corrupted when it was automatically downloaded by TractSeg. However, the segfault error persisted. I do not know if it would be a problem concerning memory access. I am using Python 3.9.13, FSL 6.0.6.2, pytorch 1.13.1, numpy 1.21.5, and MacOs 13.6.6. 

Thank you for your support!

**Comment by wasserth** [comment]
Unfortunately a "segmentation fault" error can mean anything. Therefore, I can not really tell what the problem might be. Generally speaking TractSeg is most optimised for Linux. So if you have a Linux / Ubuntu machine available you can try to run on this again.

**Comment by giovannividetta** [comment]
Hi, @wasserth!

I apologise if I re-write to you. I followed your suggestion and I tried to run TractSeg on Windows 10 by using Ubuntu 22.04.3 LTS. However, althought  "segmentation fault" error has been resolved, now the step of "Procesing direction (2 of 3)" is killed by the script. I reported the command line on the terminal:

brainmilab@brainmilab:~/tratto_gio/examples$ TractSeg -i Diffusion.nii.gz --raw_diffusion_input
Creating brain mask...
Warning: An input intended to be a single 3D volume has multiple timepoints. Input will be truncated to first volume, but this functionality is deprecated and will be removed in a future release.
Creating peaks (1 of 3)...
Creating peaks (2 of 3)...
Creating peaks (3 of 3)...
Loading weights from: /home/brainmilab/.tractseg/pretrained_weights_tract_segmentation_v3.npz
Processing direction (1 of 3)
100%|█████████████████████████████████████████████████████████████████████████████████| 144/144 [00:27<00:00,  5.28it/s]Processing direction (2 of 3)
  0%|                                                                                           | 0/144 [00:00<?, ?it/s]Killed

Is it a common error or am I making any mitake?
Thanks again for your time and support!

**Comment by giovannividetta** [comment]
I solved the issue by running TractSeg algorithm on Linux machine.
I reported here the dependencies used/installed.

Ubuntu 24.04.1 LTS
RAM memory: 16 GB
Python 3.9.19
Package                  Version
------------------------ -----------
aiohappyeyeballs         2.4.0
aiohttp                  3.10.5
aiosignal                1.3.1
async-timeout            4.0.3
attrs                    24.2.0
certifi                  2024.8.30
charset-normalizer       3.3.2
contourpy                1.3.0
cycler                   0.12.1
dataclasses-json         0.6.7
deepdiff                 8.0.1
Deprecated               1.2.14
dipy                     1.9.0
filelock                 3.16.1
fonttools                4.53.1
frozenlist               1.4.1
fsspec                   2024.9.0
fury                     0.11.0
future                   1.0.0
h5py                     3.11.0
idna                     3.10
importlib_resources      6.4.5
Jinja2                   3.1.4
joblib                   1.4.2
kiwisolver               1.4.7
lazy_loader              0.4
MarkupSafe               2.1.5
marshmallow              3.22.0
matplotlib               3.9.2
mpmath                   1.3.0
multidict                6.1.0
mypy-extensions          1.0.0
networkx                 3.2.1
nibabel                  5.2.1
numpy                    1.26.4
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.20.5
nvidia-nvjitlink-cu12    12.6.68
nvidia-nvtx-cu12         12.1.105
orderly-set              5.2.2
packaging                24.1
pandas                   2.2.2
pillow                   10.4.0
pip                      24.2
psutil                   6.0.0
pygltflib                1.16.2
pyparsing                3.1.4
python-dateutil          2.9.0.post0
pytz                     2024.2
requests                 2.32.3
scikit-learn             1.5.2
scipy                    1.13.1
seaborn                  0.13.2
setuptools               75.1.0
setuptools-scm           8.1.0
six                      1.16.0
sympy                    1.13.3
threadpoolctl            3.5.0
tomli                    2.0.1
torch                    2.4.1
tqdm                     4.66.5
TractSeg                 2.9
triton                   3.0.0
trx-python               0.3
typing_extensions        4.12.2
typing-inspect           0.9.0
tzdata                   2024.1
urllib3                  2.2.3
vtk                      9.3.1
wheel                    0.44.0
wrapt                    1.16.0
xvfbwrapper              0.2.9
yarl                     1.11.1
zipp                     3.20.2
